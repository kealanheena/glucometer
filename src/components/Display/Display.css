.display-container {
  align-items: center;
  background: var(--display-background-color);
  display: flex;
  height: var(--display-height);
  padding: 0 4%;
  margin: 4%;
  width: var(--display-width);
}

.display-value {
  color: var(--display-text-color);
  font-size: var(--display-text-size);
  font-family: var(--main-font);
  font-weight: var(--display-text-weight);
  text-align: center;
  overflow: hidden;
}

.unit-display {
  font-size: 0.4em;
}

.unit-input {
  padding: 2%;
  text-align: right;
  font-size: 0.5em;
}