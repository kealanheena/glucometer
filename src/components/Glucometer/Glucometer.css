.glucometer-container {
  background-color: var(--glucometer-background-color);
  height: var(--glucometer-height);
  width: var(--glucometer-width);
}