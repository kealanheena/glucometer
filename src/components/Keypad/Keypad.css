.keypad-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: var(--keypad-height);
  padding: 4%;
  margin: 4%;
  width: var(--keypad-width);
}

.button-container {
  display: flex;
  flex-direction: row;
  height: 20%;
  width: 100%;
}
