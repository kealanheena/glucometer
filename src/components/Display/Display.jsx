import React from 'react';
import PropTypes from 'prop-types';
import './Display.css'

const Display = ({ displayValue }) => (
  <div className="display-container">
    <div className="display-value">
      { displayValue }
    </div>
  </div>
);

Display.propTypes = { displayValue: PropTypes.any.isRequired };

export default Display;
