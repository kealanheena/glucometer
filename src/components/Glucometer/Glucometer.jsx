import React, { Component } from 'react';
import './Glucometer.css'
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';

class Glucometer extends Component {
  state = {
    // value to be displayed in <Display />
    displayValue: 'Welcome to the online carb counter',
    // value to be displayed in the <Keys />
    buttons: ['Get Bolus'],
    // ratio used to calculate your bolus(amount of insulin needed)
    // ratio: '',
    // amount of carbs you are eating in grams
    // carbs: ''
  }

  calculateBolus = () => {
    console.log('calculateBolus')
  }

  updateDisplay = () => {
    console.log('updateDisplay')
  }

  render = () => {
    // unpack the component state by using Object Destructuring
    const { displayValue, buttons } = this.state;
    return (
      <div className="glucometer-container">
        <Display displayValue={displayValue} />
        <Keypad buttons={buttons} />
      </div>
    )
  }
}

export default Glucometer;
