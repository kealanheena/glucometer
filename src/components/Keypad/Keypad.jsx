import React from 'react';
import PropTypes from 'prop-types';
import Key from '../Key/Key';
import './Keypad.css';

const Keypad = ({ 
  buttons, 
  calculateBolus,
}) => {

  // const buttonKeys = buttons.map(button => <p key={button}>{button}</p>);

  return (
    <div className="keypad-container">
      <div className="button-key">
        <Key 
          keyAction={calculateBolus} 
          keyValue='Calculate Bolus'
        />
      </div>
    </div>
  );
}

Keypad.propTypes = {
  buttons: PropTypes.array.isRequired,
}

export default Keypad;