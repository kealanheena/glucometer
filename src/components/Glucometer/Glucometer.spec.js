import React from 'react';
import { shallow, mount } from 'enzyme';
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';
import Glucometer from './Glucometer';

describe('Glucometer', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Glucometer />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the Display and Keypad Components', () => {
    expect(wrapper.containsAllMatchingElements([
      <Display displayValue={wrapper.instance().state.displayValue}/>,
      <Keypad buttons={wrapper.instance().state.buttons}/>
    ])).toEqual(true);
  });
});

describe('mounted Glucometer', () => {
  let wrapper;

  beforeEach(() => wrapper = mount(<Glucometer />));

  it('calls calculateBolus when the submit key is clicked', () => {
    const spy = jest.spyOn(wrapper.instance(), 'calculateBolus');
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalledTimes(0);
    wrapper.find('.key-container').simulate('click');
    expect(spy).toHaveBeenCalledTimes(1);
  });
});