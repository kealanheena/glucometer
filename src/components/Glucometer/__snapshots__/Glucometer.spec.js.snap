// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Glucometer should render correctly 1`] = `
<div
  className="glucometer-container"
>
  <Display
    displayValue={
      <div>
        <p>
          carbs:
        </p>
        <input
          className="unit-input"
          max="240"
          min="0"
          type="number"
        />
        <em
          className="unit-display"
        >
          g
        </em>
        <p>
          1 u :
        </p>
        <input
          className="unit-input"
          max="55.4"
          min="0"
          type="number"
        />
        <em
          className="unit-display"
        >
          mmol/L
        </em>
      </div>
    }
  />
  <Keypad
    calculateBolus={[Function]}
  />
</div>
`;
